kafka-connect-neo4j/src/main/kotlin/streams/kafka/connect/common/ConfigurationMigrator.kt

```diff
@@ -0,0 +1,164 @@
+        // Source
+        BATCH_TIMEOUT_MSECS to PropertyConverter("neo4j.batch-timeout") { convertMsecs(settings[BATCH_TIMEOUT_MSECS] as String) },
+        TOPIC to PropertyConverter("neo4j.query.topic") {settings[TOPIC] as String},
+        STREAMING_FROM to PropertyConverter("") {settings[STREAMING_FROM] as String},
+        SOURCE_TYPE to PropertyConverter("") {settings[SOURCE_TYPE] as String},
+        SOURCE_TYPE_QUERY to PropertyConverter("neo4j.query") {settings[SOURCE_TYPE_QUERY] as String},
+        STREAMING_PROPERTY to PropertyConverter("neo4j.query.streaming-property") {settings[STREAMING_PROPERTY] as String},
+        STREAMING_POLL_INTERVAL to PropertyConverter("neo4j.query.poll-interval") { convertMsecs(settings[STREAMING_POLL_INTERVAL] as String) },
+        ENFORCE_SCHEMA to PropertyConverter("") {settings[ENFORCE_SCHEMA] as String}
+    )
+
+    // Configuration properties that have user-defined keys
+    private val prefixConverterMap: Map<String, String> = mutableMapOf(
+        Neo4jSinkConnectorConfig.TOPIC_PATTERN_NODE_PREFIX to "neo4j.pattern.node.topic.",
+        Neo4jSinkConnectorConfig.TOPIC_PATTERN_RELATIONSHIP_PREFIX to "neo4j.pattern.relationship.topic.",
+        Neo4jSinkConnectorConfig.TOPIC_CYPHER_PREFIX to "neo4j.cypher.topic."
+    )
+
+    /**
+     * Migrate configuration keys from existing format to v5.1 connector format.
+     * Configuration properties containing msecs units are converted to new format.
+     *
+     * @return updated configuration key-value pairs
+     */
+    fun migrate(): Map<String, Any> {
+        val updatedConfig: MutableMap<String, String> = mutableMapOf()
+
+        settings.forEach { (originalKey, value) ->
+            val propConverter = propertyConverterMap[originalKey]
+            if (propConverter != null) {
+                val newKey = propConverter.updatedConfigKey
+                if (newKey.isBlank()) return@forEach // Configuration option found, but no new equivalent key exists
+                updatedConfig[newKey] = propConverter.migrationHandler()
+                log.debug("Migrating configuration {} to {}", originalKey, newKey)
+            } else if (prefixConverterMap.keys.any { k -> originalKey.startsWith(k) }) {
+                val prefixMatch = prefixConverterMap.keys.find { k -> originalKey.startsWith(k) }
+                prefixMatch?.let { prefix ->
+                    val replacement = prefixConverterMap[prefixMatch]
+                    replacement?.let { repl ->
+                        val newKey = originalKey.replace(prefix, repl)
+                        updatedConfig[newKey] = value
+                        log.debug("Migrating configuration prefix key {} to {}", originalKey, newKey)
+                    }
+                }
+            } else {
+                // Configuration option not declared should be copied across
+                updatedConfig[originalKey] = value
+            }
+        }
+
+        outputToLog(updatedConfig)
```

Comment:
This line is here is very important. At the same time, neither the returned type nor the method name doesn't reflect that we print the config into the log. 
I suggest removing this line and writing a separate (maybe utility) method (something like `printAsJson(o: Any)`), and passing the updated/migrated config to this method. I think think might be cleaner. 
Maybe it could be a 3-liner instead:
```
val migrated = migrator.migrate()
val json = mapper.writeValueAsString(migrated)
log.info(...)
```
which is also IMO slightly better.

Reply:
Just clarifying that you're suggesting uncoupling the logging from this class and instead doing it in, `SinkConnect/SourceConnector::stop` instead?

Reply:
done: 3-liner approach